frontend/src/views/funding/FundingOverviewView.vue

```diff
@@ -54,47 +163,211 @@
+     * This is a know issue and postponed to a future sprint.
+     */
+    async getFundingAgreements() {
+      try {
+        this.loading = true
+        this.fundingAgreements = []
+        const startDateThreshold = this.startDateThreshold()
+        const facilities = this.selectedFacility ? [this.selectedFacility] : this.userInfo.facilities
+        for (let facility of facilities) {
+          try {
+            let payload
+            if (this.selectedDateFilterType === 'Custom') {
+              payload = await FundingAgreementService.getFAByFacilityIdAndStartFromEndDates(facility.facilityId, this.startDateFrom, this.startDateTo)
+            } else {
+              payload = await FundingAgreementService.getFAByFacilityIdAndStartDateThreshold(facility.facilityId, startDateThreshold)
+            }
+            if (!payload) continue
+            await this.enrichFundingAgreementData(payload)
+            this.fundingAgreements.push(payload)
+          } catch (error) {
+            console.error(`Error getting funding agreements for facility ID ${facility.facilityId}:`, error)
+          }
+        }
+      } catch (error) {
+        console.error(`Error getting funding agreements:`, error)
+      } finally {
+        this.loading = false
+      }
+    },
+
+    async enrichFundingAgreementData(result) {
+      result.fundingAgreementType = 'Base Funding'
+      result.facility = this.facilityName(result.facilityId)
+      result.status = this.statusName(result.statusCode)
+      result.expenseAuthorityName = await this.getExpenseAuthorityName(result.applicationId)
+    },
+
+    async getExpenseAuthorityName(applicationId) {
+      try {
+        const application = await ApplicationService.getApplication(applicationId)
+        if (!application?.expenseAuthorityId) return ''
+        const expenseAuthorityUser = this.orgUsers.find(user => user.contactId === application.expenseAuthorityId)
+        return `${expenseAuthorityUser?.firstName} ${expenseAuthorityUser?.lastName}`
+      } catch (error) {
+        console.error('Error getting expense authority name:', error)
+        return ''
+      }
+    },
+
+    updateSelectedDateFilterType(value) {
+      this.selectedDateFilterType = value
+    },
+
+    startDateThreshold() {
+      switch (this.selectedDateFilterType) {
+        case '3Months':
+          return this.dateByMonthsInPast(3)
+        case '6Months':
+          return this.dateByMonthsInPast(6)
+        case 'YTD':
+          return this.dateByMonthsInPast(12)
+        default:
+          return null
+      }
+    },
+
+    dateByMonthsInPast(months) {
+      const date = new Date()
+      date.setMonth(date.getMonth() - months)
+      return date.toISOString().split('T')[0]
+    },
+
+    facilityName(facilityId) {
+      return this.userInfo.facilities.find(facility => facility.facilityId === facilityId).facilityName
+    },
+
+    statusName(statusCode) {
+      switch (statusCode) {
```

Comment:
You could just create a constant in this file mapping status codes to display names and then dynamically lookup the display name by statusCode key.

Reply:
Updated.